src/codelabs/null-safety.md

```diff
@@ -0,0 +1,321 @@
+---
+title: Null safety codelab
+description: Learn about and practice writing null safe code in DartPad!
+js: [{url: 'https://dartpad.dev/inject_embed.dart.js', defer: true}]
+---
+{% assign useIframe = false -%}
+<?code-excerpt replace="/ *\/\/\s+ignore_for_file:[^\n]+\n//g; /(^|\n)*\/\/\s+ignore:[^\n]+\n/$1/g; /(\n[^\n]+) *\/\/\s+ignore:[^\n]+\n/$1\n/g"?>
+<?code-excerpt plaster="none"?>
+
+This codelab teaches you about Dart’s null-safe type system,
+which was introduced in Dart SDK 2.12. When you opt into null safety,
+types in your code are non-nullable by default,
+meaning that values can’t be `null` unless you say they can be.
+
+This codelab covers the following material:
+
+* Nullable and non-nullable types.
+* How and when to use the null-aware operators: `?` and `!`
+* How the `late` keyword affects variables and initialization.
+
+Using embedded DartPad editors, you can test your knowledge by
+running example code and completing exercises. To get the most out of
+this codelab, you should have some knowledge of [basic Dart syntax](/samples).
+
+{{site.alert.note}}
+  This page uses embedded DartPads to display examples and exercises.
+  {% include dartpads-embedded-troubleshooting.md %}
+{{site.alert.end}}
+
+## What is null safety?
+
+Dart's null-safe type system makes types in your code non-nullable by default.
```

Comment:
s/Dart's null-safe type system makes types in your code non-nullable by default./With sound null safety, types are non-nullable by default/